fix: make pxe check tx hash on note storage

nventuro opened the pull request:
Part of a broader set of changes re. to note and event storage. I changed note storage so that it checks that the note hash exists in the effects of the tx that we queried for, and adjusted tests accordingly (mirroring the work done in the `deliverEvent` tests by @mverzilli). I did not remove the rest of the behavior during validation, e.g. checking tree leaves, as that will be done in follow up PRs as part of e.g. F-218.

mverzilli commented on yarn-project/pxe/src/contract_function_simulator/pxe_oracle_interface.test.ts:
```diff
@@ -682,62 +736,31 @@ describe('PXEOracleInterface', () => {
-        return Promise.resolve([undefined]);
-      });
+    it('should throw if tx was mined after synced block number', async () => {
+      await setSyncedBlockNumber(BlockNumber(BlockNumber(blockNumber - 1)));
```
```suggestion
      await setSyncedBlockNumber(BlockNumber(blockNumber - 1));
```

mverzilli commented on yarn-project/pxe/src/contract_function_simulator/pxe_oracle_interface.ts:
```diff
@@ -680,6 +680,20 @@ export class PXEOracleInterface implements ExecutionDataProvider {
     const uniqueNoteHash = await computeUniqueNoteHash(noteNonce, await siloNoteHash(contractAddress, noteHash));
     const siloedNullifier = await siloNullifier(contractAddress, nullifier);
 
+    const txEffect = await this.aztecNode.getTxEffect(txHash);
+    if (!txEffect) {
+      throw new Error(`Could not find tx effect for tx hash ${txHash}`);
+    }
+
+    if (txEffect.l2BlockNumber > syncedBlockNumber) {
+      throw new Error(`Could not find tx effect for tx hash ${txHash} as of block number ${syncedBlockNumber}`);
+    }
```
nit: maybe we can extract this to a `#getTxEffect(txHash)` and use it also from `deliverEvent`

mverzilli reviewed (approved):
LGTM!

nventuro committed:
fix: make pxe check tx hash on note storage

Part of a broader set of changes re. to note and event storage. I changed note storage so that it checks that the note hash exists in the effects of the tx that we queried for, and adjusted tests accordingly (mirroring the work done in the `deliverEvent` tests by @mverzilli). I did not remove the rest of the behavior during validation, e.g. checking tree leaves, as that will be done in follow up PRs as part of e.g. F-218.
